pkg/utils/builders.go

```diff
@@ -172,24 +171,13 @@ func buildJobSpec(pod *podtemplatespec.Builder) (*jobspec.Builder, error) {
 }
 
 // BuildJob will build the JobObject for creation
-func (experiment *ExperimentDetails) buildJob(pod *podtemplatespec.Builder, jobspec *jobspec.Builder) (*batchv1.Job, error) {
+func (expDetails *ExperimentDetails) buildJob(jobspec *jobspec.Builder) (*batchv1.Job, error) {
 	jobObj, err := job.NewBuilder().
 		WithJobSpecBuilder(jobspec).
-		WithAnnotations(experiment.Annotations).
-		WithName(experiment.JobName).
-		WithNamespace(experiment.Namespace).
-		WithLabels(experiment.ExpLabels).
+		WithAnnotations(expDetails.Annotations).
+		WithName(expDetails.JobName).
+		WithNamespace(expDetails.Namespace).
+		WithLabels(expDetails.ExpLabels).
 		Build()
 	return jobObj, err
 }
-
-// GetValueFromDownwardAPI returns the value from downwardApi
-func GetValueFromDownwardAPI(apiVersion string, fieldPath string) v1.EnvVarSource {
```

Comment:
Is this func used by an external package?
If it is not being used, it can be safely deleted (or, it should be private).

Reply:
Interestingly, it is being used in [container-kill](https://github.com/litmuschaos/litmus-go/blob/adc848e0920f22be11b3dc20881d017a23ddb2d8/chaoslib/litmus/container-kill/lib/container-kill.go#L282) & [network-chaos](https://github.com/litmuschaos/litmus-go/blob/adc848e0920f22be11b3dc20881d017a23ddb2d8/chaoslib/litmus/network-chaos/lib/network-chaos.go#L286) experiments as well. But has been rewritten. Thoughts @ispeakc0de ?

Reply:
It is used to pass the `POD_NAME`(ExperimentPodName) inside [builder](https://github.com/litmuschaos/chaos-runner/blob/master/pkg/utils/builders.go#L73)

Reply:
either way is fine. We can use this function to get the `v1.EnvVarSource` object instead of writing inline code inside the `getEnvFromMap` function or we can remove this as it is only used at a single place(just to get POD_NAME) and we can add this function later if downward API env required at multiple places.

Reply:
We can go w/ status quo - i.e., present changes then